mbed_greentea/mbed_test_api.py

```diff
@@ -118,119 +114,15 @@ def run_host_test(image_path,
-        """ Default process used to observe stdout of another process as console input from MUT
-        """
-        def __init__(self, proc):
-            Thread.__init__(self)
-            self.proc = proc
-            self.queue = Queue()
-            self.daemon = True
-            self.active = True
-            self.start()
-
-        def run(self):
-            while self.active:
-                c = self.proc.stdout.read(1)
-                self.queue.put(c)
-
-        def stop(self):
-            self.active = False
-
-            # Try stopping mbed-host-test
-            try:
-                self.proc.stdin.close()
-            finally:
-                pass
-
-            # Give 5 sec for mbedhtrun to exit
-            ret_code = None
-            for i in range(5):
-                ret_code = self.proc.poll()
-                # A None value indicates that the process hasn't terminated yet.
-                if ret_code is not None:
-                    break
-                sleep(1)
-
-            if ret_code is None:            # Kill it
-                print 'Terminating mbed-host-test(mbedhtrun) process (PID %s)' % self.proc.pid
-                try:
-                    self.proc.terminate()
-                except Exception as e:
-                    print "ProcessObserver.stop(): %s" % str(e)
-
-    def get_char_from_queue(obs):
-        """ Get character from queue safe way
-        """
+    def run_command(cmd):
+        """! Runs command and prints proc stdout on screen """
         try:
-            c = obs.queue.get(block=True, timeout=0.5)
-            # signals to queue job is done
-            obs.queue.task_done()
-        except Empty:
-            c = None
-        except:
-            raise
-        return c
-
-    def filter_queue_char(c):
-        """ Filters out non ASCII characters from serial port
-        """
-        if ord(c) not in range(128):
-            c = ' '
-        return c
-
-    def get_auto_property_value(property_name, line):
-        """! Scans auto detection line from MUT and returns scanned parameter 'property_name'
-        @details Host test case has to print additional properties for test to be set up
-        @return Returns string or None if property search failed
-        """
-        result = None
-        if re.search("HOST: Property '%s'"% property_name, line) is not None:
-            property = re.search("HOST: Property '%s' = '([\w\d _]+)'"% property_name, line)
-            if property is not None and len(property.groups()) == 1:
-                result = property.groups()[0]
-        return result
-
-    # Detect from where input should be taken, if no --digest switch is specified
-    # normal test execution can be performed
+            p = Popen(cmd,
+                    stdout=PIPE,
+                    stderr=STDOUT)
+        except OSError as e:
+            print "mbedgt: run_command(%s) ret= %d failed: "% (cmd, int(ret)), str(e), e.child_traceback
```

Comment:
When in exception we would like to use minimal IO to error message. Just to be clear, ```gt_logger``` is the way to go but in exception handler hard to say if ```gt_logger``` can be used.
Just to print less noise during trace IMHO.